runelite-client/src/main/java/net/runelite/client/plugins/playerindicators/PlayerIndicatorsPlugin.java

```diff
@@ -249,6 +254,64 @@ private String decorateTarget(String oldTarget, Decorations decorations)
 		return newTarget;
 	}
 
+	/*
+		Changes the player name for the trade window title to be green if they are a friend.
+	 */
+
+	@Subscribe
+	public void onScriptPostFired(ScriptPostFired event)
+	{
+		int scriptId = event.getScriptId();
+
+		if (scriptId == ScriptID.TRADE_MAIN_INIT)
+		{
+			clientThread.invokeLater(() ->
+			{
+				Widget tradeTitle = client.getWidget(WidgetInfo.TRADE_WINDOW_HEADER);
+				String header = tradeTitle.getText();
+				String playerName = header.substring("Trading with: ".length());
+				header = header.substring(0, "Trading with: ".length());
+
+				Player[] players = client.getCachedPlayers();
+				Player targetPlayer = null;
+
+				for (Player player : players)
+				{
+					if (player != null && player.getName().equals(playerName))
+					{
+						targetPlayer = player;
+					}
+				}
+
+				if (targetPlayer == null)
+				{
+					return;
+				}
+
+				if (targetPlayer.isFriendsChatMember())
+				{
+					playerName = ColorUtil.wrapWithColorTag(playerName, config.getFriendsChatMemberColor());
+				}
+				else if (targetPlayer.isFriend())
+				{
+					playerName = ColorUtil.wrapWithColorTag(playerName, config.getFriendColor());
+				}
+				else if (targetPlayer.isClanMember())
+				{
+					playerName = ColorUtil.wrapWithColorTag(playerName, config.getClanMemberColor());
+				}
+				else if (targetPlayer.getTeam() > 0 && client.getLocalPlayer().getTeam() == targetPlayer.getTeam())
+				{
+					playerName = ColorUtil.wrapWithColorTag(playerName, config.getTeamMemberColor());
+				}
```

Comment:
```java
Color playerColor = null;
if (targetPlayer.isFriendsChatMember())
{
	playerColor = config.getFriendsChatMemberColor();
}
// ...

if (playerColor != null)
{
	tradeTitle.setText(header + ColorUtil.wrapWithColorTag(playerName, playerColor);
}
```

Reply:
Also, this does not use the same color priority as the rest of player highlighting. (overlay, menu entry recoloring) It might be worth centralizing that to a method somewhere so we can avoid rewriting it over and over.